blog/2022-05-12-s6-overlay-base-images.md

```diff
@@ -43,3 +43,17 @@ You have to tweak your [AppArmor profile](/docs/add-ons/presentation#apparmor) t
   /run/{,**} rwk,
   /dev/tty rw,
 ```
+
+## `host_pid` option
+
+Addons running without protection mode enabled can set `host_pid: true` in their configuration. As described in the [documentation](https://developers.home-assistant.io/docs/add-ons/configuration#optional-configuration-options):
+
+> Allow to run container on host PID namespace. Works only for not protected add-ons.
+
+This is a problem because S6 expects to be PID 1 (it's literally in the [tagline](https://github.com/just-containers/s6-overlay#s6-overlay-)) and that's impossible when using the host PID namespace.
+
+In V2, S6 didn't actually check that it was running as PID 1. This is why it "worked" when in this mode in the past (although it required some [workarounds](https://github.com/hassio-addons/addon-glances/blob/8575d7903ef4c0a7c49e9ab32e0536bd2eb12dd6/glances/rootfs/bin/s6-nuke) to keep s6 from breaking systems when running this way). In V3 S6 checks that it is actually PID 1 and refuses to start otherwise.
+
+There's no easy solution to this, the option simply won't work in S6 V3. As a temporary fix you can switch to using the corresponding [community add-ons base image](https://github.com/hassio-addons?q=base&type=all&language=&sort=). I must stress that this is temporary though, eventually those will update too.
+
+Long-term you should not use s6 overlay in your addon as its not designed for this. You should use stock [alpine](https://hub.docker.com/search?q=alpine&type=image), [debian](https://hub.docker.com/_/debian) or [ubuntu](https://hub.docker.com/_/ubuntu) as your base image instead and add what you need to the image to make your addon work.
```

Comment:
Or, override init so S6 won't initialize.

Reply:
Is that a better suggestion then this? That's basically an alternate way of not using S6. I guess since you still get tempio and bashio?

Reply:
Yeah, overwrite/reset default I guess is the best way to describe it. But I mean that option is not used a lot

Reply:
Ok. Probably still going to suggest the quick fix is the community addons image. Turning off S6 also means refactoring away from using cont-init.d and services.d so you have work to do with that option. But can propose that as the long-term solution.

Reply:
> Probably still going to suggest the quick fix is the community addons image.

Please don't. People should fix it, not workaround. This blog will be there for a long time, and the change to the community add-on will happen anyways (thus is not a path to take).

Reply:
Ok, removed

Reply:
Thanks! That prevents people starting to complain on my end when I migrate 👍